dttools/src/set.h

```diff
@@ -47,16 +48,31 @@ while(element = set_next_element(s)) {
 
 struct set *set_create(int buckets);
 
+/** Duplicate a set from an existing set.
```

Comment:
For all these new functions, please add a warning in the documentation saying that pointers are not duplicated, and that one has to watch for double frees.